### PR TITLE
Add AdditionalTrustBundle to NutanixPrismEndpoint

### PR DESCRIPTION
**Pull request by thunderboltsid**
AdditionalTrustBundle is a PEM-encoded certificate bundle to be used in addition to system trust store.

### CODE REVIEW

**Review by heiko-koehler — DISMISSED** [review]
LGTM

Thanks @sid for adding code generator for credentials. I think I've generated code manually before.

File: environment/credentials/types.go [hunk 1]
```diff
@@ -96,6 +96,12 @@ type NutanixPrismEndpoint struct {
 	// +kubebuilder:default=false
 	// +optional
 	Insecure bool `json:"insecure"`
+	// AdditionalTrustBundle is a PEM encoded x509 cert for the RootCA that was used to create the certificate
+	// for a Prism Central that uses certificates that were issued by a non-publicly trusted RootCA. The trust
+	// bundle is added to the cert pool used to authenticate the TLS connection to the Prism Central.
+	// +kubebuilder:validation:Type=string
+	// +optional
+	AdditionalTrustBundle string `json:"additionalTrustBundle,omitempty"`
```
Comment (yannickstruyf3) [inline environment/credentials/types.go:104]: Can we change this from string to a struct similar to NutanixCredentialReference? This will allow us to add additional data sources for the trust bundle. An example could be a ConfigMap

File: environment/providers/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -15,9 +15,25 @@ import (
...
+	}
+	if prov.prismEndpoint.AdditionalTrustBundle.Kind == credentials.NutanixTrustBundleKindString {
+		return prov.prismEndpoint.AdditionalTrustBundle.Data, nil
+	}
+	trustBundleRef := prov.prismEndpoint.AdditionalTrustBundle
+	cm, err := prov.cmInformer.Lister().ConfigMaps(trustBundleRef.Namespace).Get(trustBundleRef.Name)
+	if err != nil {
+		return "", err
+	}
+	return cm.Data["ca.crt"], nil
```
Comment (thunderboltsid) [inline environment/providers/kubernetes/kubernetes.go:34]: any opinions on the key here?

File: environment/credentials/types.go (hunk 1 above)
Comment (thunderboltsid) [reply environment/credentials/types.go]: updated the type; PTAL

**Review by heiko-koehler — APPROVED** [review]

**Review by tuxtof — APPROVED** [review]

File: environment/providers/kubernetes/kubernetes_test.go [hunk 3]
```diff
@@ -37,11 +37,27 @@ func runSecretInformer(ctx context.Context, clientset kubernetes.Interface) core
 	return secretInformer
 }
 
+func runCMInformer(ctx context.Context, clientset kubernetes.Interface) coreinformers.ConfigMapInformer {
```
Comment (deepakm-ntnx) [inline environment/providers/kubernetes/kubernetes_test.go:40]: could you please add comments on why is this needed?

**Comment by deepakm-ntnx** [comment]
It would be good to add more test details in the description. also more details on the overall change would be good.

File: environment/providers/kubernetes/kubernetes_test.go (hunk 3 above)
Comment (thunderboltsid) [reply environment/providers/kubernetes/kubernetes_test.go:40]: The Kubernetes environment provider fetches the certificate from a configmap
